File: internal/model/game.go
package model

import (
	"time"
)

type Game struct {
	UUID         string    
	CreatedAt    time.Time 
	UpdatedAt    time.Time 
	Name         string    
	GameType     string    
	BoardState   []byte    
	NextPlayerID int       
	Player1Piece string    
	Player2Piece string    
}

File: frontend/src/shared/types/game.ts
import type { GameToken } from "./game-token";

export type Game = {
    boardState: string;
    gameType: "neural_network" | "minimax";
    name: string;
    nextPlayerId: number;
    player1Piece: GameToken;
    player2Piece: GameToken;
    terminalState: number;
    uuid: string;
}
